Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -1001,15 +1001,75 @@ void Element::set_scroll_top(double y)
     block_container->set_scroll_offset(scroll_offset);
 }
 
+// https://w3c.github.io/csswg-drafts/cssom-view/#dom-element-scrollwidth
 int Element::scroll_width() const
 {
-    dbgln("FIXME: Implement Element::scroll_width() (called on element: {})", debug_description());
+    // 1. Let document be the element's node document.
+    auto const& document = this->document();
+
+    // 2. If document is not the active document, return zero and terminate these steps.
+    if (!document.is_active())
+        return 0;
+
+    // 3. Let viewport width be the width of the viewport excluding the width of the scroll bar,
+    //    if any, or zero if there is no viewport.
+    // FIXME: Check if there is a viewport
```

Comment:
If `document.browsing_context()` returns null, there is no viewport. So we can use that here.

Reply:
Thank you! I added this check. As a separate question, how do we calculate the viewport scrolling area width if there is no viewport? The spec is not very clear on this point.